packets.go

```diff
@@ -616,22 +617,31 @@ func (rows *textRows) readRow(dest []driver.Value) error {
 		pos += n
 		if err == nil {
 			if !isNull {
-				if !mc.parseTime {
```

Comment:
There shouldn't be any edits necessary in `readRow` at all.
MySQL has basically two protocols: A binary protocol (where the bug seems to appear) and a text protocol. `readRow` is for the text protocol. Here the MySQL server just sends everything (hopefully) correctly formated as a string.

We just check for `mc.parseTime` here because with the DSN param `parseTime=true` the users asks us to return time values as `time.Time`, not as a string. Since the database/sql package can't parse it, we have to help here. Also kind of a dirty workaround.

Please try if it works properly if you don't change here anything at all. According to your bug description it should work.